### ISSUE TITLE
Glossary definition modal is not fully accessible

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#3507]
Pressbooks processes glossary shortcodes by printing a button. See the ARIA pattern for button: https://www.w3.org/WAI/ARIA/apg/patterns/button/. When clicked, this button causes a div containing the definition to become visible on the screen. When the user presses escape or clicks outside of the definition modal, the definition disappears. The patterns resembles the dialog (modal) pattern described here: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/, but with a few problems. 

The button pattern suggests:
> Following button activation, focus is set depending on the type of action the button performs. For example:
If activating the button opens a dialog, the focus moves inside the dialog. 

In our case, the user's focus does not shift to the glossary definition, which poses problems as users of assistive technology generally cannot access or hear the term's definition. 

> The button has an accessible label. By default, the accessible name is computed from any text content inside the button element. However, it can also be provided with [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) or [aria-label](https://w3c.github.io/aria/#aria-label).

> If a description of the button's function is present, the button element has [aria-describedby](https://w3c.github.io/aria/#aria-describedby) set to the ID of the element containing the description.

Our button does not have a description of the function, but does use aria-descriedby. The aria-describedby points to the ID of the glossary__tooltip which contains the definition. It might be better to have an accessible description that describes the button's function (`display term definition` or similar)?

We probably also want to provide a clear indicator or 'close' button inside the tooltip definition div which allows the user to exit modal/popup and return to the prior page navigation with their focus on the button they initially launched. 

We may want to add aria-haspopup="dialog" to the button, to make it clear that clicking the button will open a popup which includes dialog? Not sure if this is important or will make a difference. 

This description of accessible popups: https://theadminbar.com/accessibility-weekly/how-to-make-accessible-popups/ and accompanying CodePen: https://codepen.io/amberhinds/pen/gOBpBBo provide an accessible pattern that could be adapted for our glossary terms.

## Notes
The user who reported this concern to us provided this video: https://drive.google.com/file/d/1Zl8zSoQ0dlHyCVhCqYjZpsigHudG_fnz/view and shared some suggestions:

>  Please note, these hover option “definitions” are not links they are buttons.  They look like links on the page, but they are programmed and announced as buttons. I just want to add that this distinction of links versus buttons matters because it affects user expectations. If a screen reader announces an element as a “link” or “button,” users have expectations about what will happen when they click that element. If something else happens, this can be disorienting.

> Also, from the screen reader perspective the pop window appears to never receive focus once it is opened up by the user engaging with the link text coded as a button with their keyboard. The easy thing to try first is to code them as links and see how that works and have somebody test it with a screen reader. Yet, they need to code those links so that focus goes into the popup window content and that a screen reader user can easily close out those windows and have focus continues to read exactly where they left off.
 
> If that doesn’t work, then maybe they need to be linked directly to the glossary page where that definition resides.  The key is to make sure that a screen reader user can return to the exact same spot where they left off reading when they opened the glossary. And when I say first trying to change the buttons to links so that you have the same premise of a pop-up window with that definition. And it still stands that if that is something that will work, a screen reader user is able to easily close out that pop-up window and continue reading from where they left off so that they don’t lose any understanding of the content that is being read out to them.

**Comment by SteelWagstaff** [comment]
@greatislander when implementing a solution, I'd like us to also address https://github.com/pressbooks/buckram/issues/372, probably via this proposed solution: https://github.com/pressbooks/buckram/issues/372

**Comment by tw77** [comment]
Testing this on dev, I see that the solution works as expected in Chrome.

I also see that https://github.com/pressbooks/buckram/issues/372 is fixed.

Testing on dev with Firefox though, I'm unable to tab to the glossary term in webbook text, which was possible before (compare https://dev.pressbooks.network/testampersands/chapter/sentence/ vs. the old https://university.pressbooks.pub/testnew/chapter/chapter-1/ in Firefox)

**Comment by tw77** [comment]
Tabbing actually works as expected in Firefox on dev, it just requires enabling the browser setting "Use the tab key to move focus between form controls and links" from Preferences > General, under Browsing.

It is also working as expected in Safari, where a similar browser setting is required in Preferences > Advanced, under Accessibility.

**Comment by tw77** [comment]
Working as expected and ready for release.
